File: emailsender_api.py
from flask import Flask, request, jsonify
from Verifier.email import EmailSender

app = Flask(__name__)
email_sender = EmailSender()

@app.route('/')
def index():
    return jsonify({'message': 'Email Sender API is running...'}), 200

@app.route('/send-email', methods=['POST'])
def send_email():
    data = request.json
    
    reciever = data.get('email_reciever')
    msg_subject = data.get('msg_subject', 'Email Sender')
    msg_body = data.get('msgf_body', 'Test Message.')
    sender_name = data.get('sender_name', 'EmailSender')
    
    if not reciever:
        return jsonify({'error':'reciever email is required.'}), 400
    
    try:
        email_sender.send_email(reciever, msg_subject, msg_body, sender_name)
        return jsonify({'message':'Email send successfully!'}), 200
    except Exception as e:
        return jsonify({'error':str(e)}), 500
    
if __name__ == '__main__':
    app.run(debug=True)

File: APITest/EmailSender_API.js
// install required package.
// $ npm install axios

const axios = require('axios');

async function sendEmail(reciever) {
    const API_URL = "https://emailsender000.pythonanywhere.com/send-email";

    const data = {
        email_reciever: reciever,
        msg_subject: 'Email Test',
        msg_body: 'This is the sample message using EmailSender API.',
        sender_name: 'EmailSender'
    };

    try {
        const response = await axios.post(API_URL, data, {
            headers: {
                'Content-Type': 'application/json'
            }
        });
        console.log('success: ', response.data);
    }
    catch (error) {
        console.log('error: ', error.response ? error.response.data : error.message);  
    }
}

sendEmail('christiang03112003@gmail.com');
